src/main/java/tech/autodirect/api/upstream/SensoRateQuery.java

```diff
@@ -0,0 +1,26 @@
+package tech.autodirect.api.upstream;
+
+/**
+ * The SensoRateQuery class represents a single query to the SensoApi/rate api and contains
+ * all the information that will be passed as arguments when querying.
+ */
+public class SensoRateQuery {
```

Comment:
I think we can simply use a Map for this, but making a dedicated class is also okay

Reply:
@Shalev-Lifshitz

Reply:
Good point. The examples I looked at with Gson used a dedicated class, I'll try it with a HashMap. @sammdu